src/Behat/Context/EntityContext.php

```diff
@@ -314,44 +314,82 @@ protected function checkEntityValues($entity, $fields) {
     return $errors;
   }
 
+  /**
+   * Make replacement on given string when this one contains entity-replacement.
+   *
+   * [entity-replacement:user:mail:behat@metadrop.net:uid]
+   *
+   * @param string $value
+   *   String that contains a single or multiple entity-replacement tokens.
+   *
+   * @return string|bool
+   *   Replaced or untouched string or FALSE otherwise.
+   *
+   * @throws \Exception
+   */
+  protected function entityTokensReplacements($value) {
+    $entity_tokens = [];
+    $replacements = [];
+    if (strpos($value, 'entity-replacement') === 0) {
+      $entity_tokens = [$value];
+    }
+    else {
+      $matches = [];
+      preg_match_all('#\[entity-replacement:?([^]]*)\]#', $value, $matches);
+      $entity_tokens = reset($matches);
+    }
+
+    if (empty($entity_tokens)) {
+      return FALSE;
+    }
+
+    // Get entity type list.
+    $entity_types = $this->getCore()->getEntityTypes();
+
+    foreach ($entity_tokens as $entity_token) {
+      $token_pieces = explode(':', str_replace(['[', ']'], ['', ''], $entity_token));
+      array_shift($token_pieces);
+      $entity_type = $token_pieces[0];
```

Comment:
1) It should be checked that the token pieces are correctly formed exist before accessing their keys.
2) The code can be reduced from:
```
$entity_type = $token_pieces[0];
$field_key = $token_pieces[1];
$field_value = $token_pieces[2];
$destiny_replacement = $token_pieces[3];
```

to:

```php
[$entity_type, $field_key, $field_value, $destiny_replacement] = $token_pieces;
```